tests/MongoDB.Driver.Tests/Specifications/client-side-encryption/prose-tests/ClientEncryptionProseTests.cs

```diff
@@ -721,6 +720,217 @@ void TestCase(ClientEncryption testCaseClientEncription, BsonDocument masterKey,
+                                });
+                        }
+
+                        collCollection = GetCollection(client_encrypted, __collCollectionNamespace);
+                        var findResult = Find(collCollection, BsonDocument.Parse("{ _id : 0 }"), async).Single();
+                        findResult.Should().Be(BsonDocument.Parse($"{{ _id : 0, encrypted : '{value}' }}"));
+                        var events = eventCapturer.Events.ToList();
+                        AssertEvents(events.OfType<CommandStartedEvent>(), clientEncryptedEventsExpectation);
+                        if (clientKeyVaultEventsExpectation != null)
+                        {
+                            AssertEvents(clientKeyVaultEventCapturer.Events.OfType<CommandStartedEvent>(), clientKeyVaultEventsExpectation);
+                        }
+
+                        AssertNumberOfClients(events.OfType<ClusterOpeningEvent>());
+                    }
+                }
+
+                IMongoClient GetKeyVaultMongoClientByKey()
+                {
+                    switch (keyVaultMongoClientKey)
+                    {
+                        case "client_keyvault":
+                            return client_keyvault;
+                        default:
+                            return null;
+                    }
+                }
+            }
+
+            void AssertEvents(IEnumerable<CommandStartedEvent> events, string[] expectedEventsDetails)
+            {
+                for (int i = 0; i < expectedEventsDetails.Length; i++)
+                {
+                    var arguments = expectedEventsDetails[i].Split(';');
+                    (string CommandName, string Database) expectedEventDetails = (arguments[0], arguments[1]);
+                    var @event = events.ElementAt(i);
+                    @event.DatabaseNamespace.DatabaseName.Should().Be(expectedEventDetails.Database);
+                    @event.CommandName.Should().Be(expectedEventDetails.CommandName);
+                }
+                events.Count().Should().Be(expectedEventsDetails.Count());
+            }
+
+            void AssertNumberOfClients(IEnumerable<ClusterOpeningEvent> events)
+            {
+                events.Count().Should().Be(expectedNumberOfClients);
+            }
+        }
+
+        public static IEnumerable<object[]> DeadlockTest_MemberData()
+        {
+            var testCases = new[]
+            {
+                CaseWithAsync(
+                    name: "case 1",
+                    maxPoolSize: 1,
+                    bypassAutoEncryption: false,
+                    keyVaultMongoClient: null,
+                    expectedNumberOfClients: 2,
+                    clientEncryptedEventsExpectation:
+                    new[]
+                    {
+                        "listCollections;db",
+                        "find;keyvault",
+                        "insert;db",
+                        "find;db"
+                    },
```

Comment:
Alternative if you apply tuple suggestion below:
```suggestion
                    new[]
                    {
                        ("listCollections", "db"),
                        ("find", "keyvault"),
                        ("insert", "db"),
                        ("find", "db")
                    },
```

Reply:
answered above